grammar.js

```diff
@@ -416,7 +423,34 @@ module.exports = grammar(C, {
       alias($.operator_cast_definition, $.function_definition),
       alias($.operator_cast_declaration, $.declaration),
       $.friend_declaration,
-      seq($.access_specifier, ':'),
+      $.access_specifier_list,
+      $.alias_declaration,
+      $.using_declaration,
+      $.type_definition,
+      $.static_assert_declaration,
+      ';',
```

Comment:
Semicolons can technically appear anywhere a statement is valid. I think you were trying to close a particular issue regarding having an extraneous semicolon (#35?), but I think we should hold off and try to figure out a way to make "empty" statements (a single semicolon) work anywhere a statement could appear (any of which would end with a semicolon), instead of only as a field declaration.

Not sure how terrible that would be for node count though :)

Reply:
Hm, maybe something like javascript's automatic semicolons could be used, only adds 7 to state count though

Reply:
Not familiar with that, but I think we should make something like an `empty_statement: $ => ';'` rule and just put that as an option in various places where other statements (including top level) can be. Not the most elegant and I haven't actually tried this so it may not really work in practice but something like that is the best way to make the grammar not choke on extra semicolons.

Reply:
https://github.com/tree-sitter/tree-sitter-javascript/blob/master/src/scanner.c, though C++ is a bit different and an empty_statement node would be just fine as well - but I think naming it `_empty_statement`  to hide the rule could be better